fe/src/main/java/org/apache/doris/alter/RollupJobV2.java

```diff
@@ -0,0 +1,736 @@
+            for (Tablet tablet : rollupIdx.getTablets()) {
+                totalReplicaNum += tablet.getReplicas().size();
+            }
+        }
+        MarkedCountDownLatch<Long, Long> countDownLatch = new MarkedCountDownLatch<Long, Long>(totalReplicaNum);
+        db.readLock();
+        try {
+            OlapTable tbl = (OlapTable) db.getTable(tableId);
+            if (tbl == null) {
+                cancel("Table " + tableId + " does not exist");
+                return;
+            }
+            Preconditions.checkState(tbl.getState() == OlapTableState.ROLLUP);
+
+            for (Map.Entry<Long, MaterializedIndex> entry : this.partitionIdToRollupIndex.entrySet()) {
+                long partitionId = entry.getKey();
+                Partition partition = tbl.getPartition(partitionId);
+                if (partition == null) {
+                    continue;
+                }
+                TStorageMedium storageMedium = tbl.getPartitionInfo().getDataProperty(partitionId).getStorageMedium();
+                MaterializedIndex rollupIndex = entry.getValue();
+
+                Map<Long, Long> tabletIdMap = this.partitionIdToBaseRollupTabletIdMap.get(partitionId);
+                for (Tablet rollupTablet : rollupIndex.getTablets()) {
+                    long rollupTabletId = rollupTablet.getId();
+                    List<Replica> rollupReplicas = rollupTablet.getReplicas();
+                    for (Replica rollupReplica : rollupReplicas) {
+                        long backendId = rollupReplica.getBackendId();
+                        Preconditions.checkNotNull(tabletIdMap.get(rollupTabletId)); // baseTabletId
+                        countDownLatch.addMark(backendId, rollupTabletId);
+                        // create replica with version 1.
+                        // version will be updated by following load process, or when rollup task finished.
+                        CreateReplicaTask createReplicaTask = new CreateReplicaTask(
+                                backendId, dbId, tableId, partitionId, rollupIndexId, rollupTabletId,
+                                rollupShortKeyColumnCount, rollupSchemaHash,
+                                Partition.PARTITION_INIT_VERSION, Partition.PARTITION_INIT_VERSION_HASH,
+                                rollupKeysType, TStorageType.COLUMN, storageMedium,
+                                rollupSchema, tbl.getCopiedBfColumns(), tbl.getBfFpp(), countDownLatch);
+                        createReplicaTask.setBaseTablet(tabletIdMap.get(rollupTabletId), baseSchemaHash);
+
+                        batchTask.addTask(createReplicaTask);
+                    } // end for rollupReplicas
+                } // end for rollupTablets
+            }
+        } finally {
+            db.readUnlock();
+        }
+
+        // send all tasks and wait them finished
+        AgentTaskQueue.addBatchTask(batchTask);
+        AgentTaskExecutor.submit(batchTask);
+        // max timeout is 30 seconds
```

Comment:
Maybe I should change it to a configuration. And default value is 30 sec.
Because I don't want to stuck in here for a long time if error happens. But for
the extreme case like you said, user can always modify the config to satisfy their case.

Reply:
Yes. configuration is OK, like create table logic.